### ISSUE TITLE
Error: velero does not cleanup volumesnapshotcontent completely

### ISSUE THREAD

**Issue by datle1** [vmware-tanzu/velero#4470]
**What steps did you take and what happened:**
I created a schedule for backup every 4h and TTL is 1 day.
After a few days, I find out some volumesnapshotcontents have not been cleaned although their backups were expired.

**What did you expect to happen:**
All volumesnapshotcontents should be deleted completely when the backup is expired

**The following information will help us better understand what's going on**:
- There was not any unnormal or error log at Velero and CSI plugin when the error occurred.
- The volumesnapshotcontents which got this issue were in deleting state with policy is `Delete`, although backup object and  volumesnapshot object of them had cleaned. Manually, I can not delete these volumesnapshotcontents unless removing finalizer `snapshot.storage.kubernetes.io/volumesnapshotcontent-bound-protection`. And, I don't know why ?
```
root@ks-k8smaster-1 ~# kubectl get volumesnapshotcontent | grep Delete
snapcontent-169b3534-4307-48fa-886f-effecd97142b   true         1073741824    Delete           cinder.csi.openstack.org   cinder-csi-snapshot   velero-vv-sw-v1-0-cn97v   13h
```
- Describe details of volumesnapshotcontent:
```
root@ks-k8smaster-1 ~# kubectl describe volumesnapshotcontent snapcontent-169b3534-4307-48fa-886f-effecd97142b
Name:         snapcontent-169b3534-4307-48fa-886f-effecd97142b
Namespace:    
Labels:       velero.io/backup-name=sch-proj1-20211222120004
Annotations:  <none>
API Version:  snapshot.storage.k8s.io/v1beta1
Kind:         VolumeSnapshotContent
Metadata:
  Creation Timestamp:             2021-12-22T12:00:24Z
  Deletion Grace Period Seconds:  0
  Deletion Timestamp:             2021-12-22T16:24:51Z
  Finalizers:
    snapshot.storage.kubernetes.io/volumesnapshotcontent-bound-protection
  Generation:  3
  Managed Fields:
    API Version:  snapshot.storage.k8s.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .:
          v:"snapshot.storage.kubernetes.io/volumesnapshotcontent-bound-protection":
      f:spec:
        .:
        f:driver:
        f:source:
          .:
          f:volumeHandle:
        f:volumeSnapshotClassName:
        f:volumeSnapshotRef:
          .:
          f:apiVersion:
          f:kind:
          f:name:
          f:namespace:
          f:resourceVersion:
          f:uid:
    Manager:      snapshot-controller
    Operation:    Update
    Time:         2021-12-22T12:00:24Z
    API Version:  snapshot.storage.k8s.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:creationTime:
        f:readyToUse:
        f:restoreSize:
        f:snapshotHandle:
    Manager:      csi-snapshotter
    Operation:    Update
    Time:         2021-12-22T12:01:28Z
    API Version:  snapshot.storage.k8s.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:labels:
          .:
          f:velero.io/backup-name:
      f:spec:
        f:deletionPolicy:
    Manager:         velero-plugin-for-csi
    Operation:       Update
    Time:            2021-12-22T16:24:51Z
  Resource Version:  30800085
  Self Link:         /apis/snapshot.storage.k8s.io/v1beta1/volumesnapshotcontents/snapcontent-169b3534-4307-48fa-886f-effecd97142b
  UID:               41a52672-4078-4a29-9bc5-60aab927b08f
Spec:
  Deletion Policy:  Delete
  Driver:           cinder.csi.openstack.org
  Source:
    Volume Handle:             becd8354-4bd7-455c-91f5-d6163f40a697
  Volume Snapshot Class Name:  cinder-csi-snapshot
  Volume Snapshot Ref:
    API Version:       snapshot.storage.k8s.io/v1beta1
    Kind:              VolumeSnapshot
    Name:              velero-vv-sw-v1-0-cn97v
    Namespace:         proj1
    Resource Version:  30711952
    UID:               169b3534-4307-48fa-886f-effecd97142b
Status:
  Creation Time:    1640174578252386000
  Ready To Use:     true
  Restore Size:     1073741824
  Snapshot Handle:  1b600a4e-47f1-4c88-80d3-59f2af1f50ab
Events:             <none>
```

**Anything else you would like to add:**
- Velero should add a few error logs when volumesnapshotcontent can not cleaned
- All volumesnapshotcontent should be cleaned up when backup is expired

**Environment:**

- Velero version: v1.7.0 (Git commit: 9e52260568430ecb77ac38a677ce74267a8c2176)
- Velero features: EnableCSI
- Kubernetes version: v1.19.9
- Cloud provider or hardware configuration: 
  - CSI Cinder controller plugin: v1.20.0, csi-snapshotter: v3.0.3.
  - Snapshot-controller: v3.0.3
  - velero-plugin-for-aws: v1.3.0
  - velero-plugin-for-csi: v0.2.0
  - Storage backend: ceph object (radosgw enable S3 API), ceph block (CSI cinder snapshot)

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by qiuming-best** [comment]
@datle1 velero-plugin-for-csi support level for CSI volume snapshotting is following upstream Kubernetes, currently, it's not a stable version.  Suggest you not use the velero-plugin-for-csi,  and in the meantime, I'll test it and try to figure out the problem.

**Comment by datle1** [comment]
> @datle1 velero-plugin-for-csi support level for CSI volume snapshotting is following upstream Kubernetes, currently, it's not a stable version. Suggest you not use the velero-plugin-for-csi, and in the meantime, I'll test it and try to figure out the problem.

- Thanks @qiuming-best for response. I see CSI volume snapshot are moved to GA from Kubernetes 1.20. It maybe mature in recent future. Velero has any roadmap for stable version of  velero-plugin-for-csi ? 
- I 'm willing to debug Velero to solve this issue together. I want to use Velero with CSI driver of Openstack Cinder.

**Comment by qiuming-best** [comment]
@datle1，thank you for providing info for us,  I just test gke csi driver(pd.csi.storage.gke.io), both volumesnapshot and volumesnapshotcontent could be deleted after backups were expired.  CSI driver of Openstack Cinder I haven't tested it, as you described, there have some problems indeed. velero has triggered the delete action, maybe the behavior of  CSI driver of Openstack Cinder or  CSI plugin was not as expected.  I'm still investigating this issue. It would be fine if you can
solve this issue together

**Comment by qiuming-best** [comment]
@dsu-igeek, Could you help to take a look at the problem

**Comment by blackpiglet** [comment]
@datle1 
Could you check with your CSI external-snapshotter's version?
There is a similar issue to this in CSI plugin repo, and it's already fixed. 
Could you please check your environment's CSI included this fix.
https://github.com/kubernetes-csi/external-snapshotter/issues/248

**Comment by blackpiglet** [comment]
@datle1 
Hi, there are some bug fixes in Velero v1.9 for CSI feature.
Would you mind to have a try with Velero and Velero CSI plugin's main image to see whether this issue is fixed?
